typed-racket-lib/typed-racket/types/subtype.rkt

```diff
@@ -705,6 +704,13 @@
                (and (not init-rest) (not init-rest*)
                     A)))]
      [_ (continue<: A t1 t2 obj)])]
+  [(case: Con (Con: t1-pre t1-post))
+   (match t2
+     [(Con: t2-pre t2-post)
+      (subtype-seq A
+                   (subtype* A t2-pre t1-pre obj)
```

Comment:
During its expansion, `subtype-seq` _adds_ an argument to the head of each operand list after the first expression (i.e. after the initial `A`), so your passing two `A` values to `subtype*` here (and in other places).

Take a gander at other usages of `subtype-seq` in this file and you'll see they seem to mysteriously omit the first argument to `subtype*` usages in the body of the `subtype-seq` (because `subtype-seq` provides it).

Reply:
I haven't built and tested this hypothesis, but it's almost certainly what is causing the `mask-accessor` issue you noticed.

The `mask` error says basically it expected a type (i.e. something with a `mask` struct property field -- defined [here](https://github.com/racket/typed-racket/blob/master/typed-racket-lib/typed-racket/rep/rep-utils.rkt#L190)), but I got `'()` (i.e. an empty `A` argument in the incorrect `subtype-seq` usage).

Reply:
i.e. it should read (assuming all the other details are correct):

```racket
(subtype-seq A
             (subtype* t2-pre t1-pre obj)
             (subtype* t1-post t2-post obj))
```